components/VoteCommentModal.tsx

```diff
@@ -142,6 +140,14 @@ const VoteCommentModal: FunctionComponent<VoteCommentModalProps> = ({
 
   const voteString = VOTE_STRINGS[vote]
 
+  useEffect(() => {
```

Comment:
I think this is probably too hacky; I really don't like it if side effects are used for business logic etcetera like this. 
What I'd rather see is the `submitVote` callback extracted to a hook and used by both this and the parent component, or simply passed as a prop to the modal by the parent.

Reply:
fair, ill work on extracting the logic somehow